analyzer-comments/elm/gotta-snatch-em-all/boringCards_use_fold.md

```diff
@@ -0,0 +1,4 @@
+# boringCards use fold
+
+The function `boringCards` can be elegantly written with a `List.foldl` (or `List.foldl`).
```

Comment:
```suggestion
The function `boringCards` can be elegantly written with a `List.foldr` (or `List.foldl`).
```

Reply:
Do I rememver rightly that foldr is more performant?

Reply:
I'm not so sure actually, I looked into it a bit and I couldn't find a crystal clear answer.

In Haskell, `foldl` is clearly bad because it uses too much memory (or something) but that's because it's lazy, and Elm is not lazy. `foldl` is conceptually easier to turn into an imperative loop because it starts computing from the head so in that sense it might be more efficient, but I think the compiler has some fail safe for `foldr`?

Reply:
Yeah, I thought foldl was implemented using foldr, for some function recursion type reason, but I can't remember the details to be honest, and it's not something that gets talked about ever, so nothing to worry about. And like you say, foldl is easier on the brain ...